src/main/java/org/openrewrite/java/spring/boot3/MaintainTrailingSlashURLMappings.java

```diff
@@ -57,11 +57,18 @@ public AtomicBoolean getInitialValue(ExecutionContext ctx) {
     public TreeVisitor<?, ExecutionContext> getScanner(AtomicBoolean acc) {
         return new JavaIsoVisitor<ExecutionContext>() {
             @Override
-            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext ctx) {
-                if (!acc.get()) {
-                    acc.set(FindWebConfigurer.find(cu));
+            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                if (!acc.get() && classDecl.getImplements() != null) {
+                    for (TypeTree impl : classDecl.getImplements()) {
+                        JavaType.FullyQualified fullyQualified = TypeUtils.asFullyQualified(impl.getType());
+                        if (fullyQualified != null &&
+                                (WEB_MVC_CONFIGURER.equals(fullyQualified.getFullyQualifiedName()) ||
+                                        WEB_FLUX_CONFIGURER.equals(fullyQualified.getFullyQualifiedName()))) {
+                            acc.set(true);
```

Comment:
We could use `break` to 100% match existing behaviour and to save some iterations.
```suggestion
                            acc.set(true);
                            break;
```

Reply:
If it is removed intentionally, no objections